src/annotator/guest.js

```diff
@@ -367,6 +352,17 @@ export default class Guest {
       this.setHighlightsVisible(showHighlights);
     });
 
+    this._sidebarRPC.on(
+      'deleteAnnotation',
+      /** @type {string} */ tag => this.detach(tag)
+    );
+
+    this._sidebarRPC.on(
+      'loadAnnotations',
+      (/** @type {AnnotationSafeData[]} */ annotations) =>
+        annotations.forEach(annotation => this.anchor(annotation))
+    );
```

Comment:
Since `anchor` is asynchronous there is a race condition here if a `deleteAnnotation` event is received for an annotation while it is being anchored. The `detach` method operates on the `this.anchors` array, but loaded annotations are not added to that until the end of the `anchor` method.

Reply:
Would an `await` here fix the race condition?

Reply:
```
    async annotations =>
        await Promise.all(
          annotations.map(annotation => this.anchor(annotation))
        )
```

Reply:
This seems to be a pre-existing issue, isn't it?

Reply:
> Would an await here fix the race condition?

No. Think about what is happening here. The proposed code will just result in the `loadAnnotations` RPC method handler receiving a promise indicating when the operation is complete. `Bridge`/`PortRPC` won't do anything with that promise though. The race here is between the `detach` method, which relies on the annotation having been added to the `this.anchors` array, and `anchor` which only adds the anchor to that array asynchronously.

The one solution would be to keep track of annotations that are currently being anchored and defer de-anchoring for them in the `deleteAnnotations` callback.

> This seems to be a pre-existing issue, isn't it?

This is true. Perhaps the expedient thing would be to note the issue in a comment and address it in a separate PR.